content/blog/kubernetes-yaml-generation/index.md

```diff
@@ -0,0 +1,208 @@
+tags:
+    - kubernetes
+    - yaml
+    - kubernetesx
+
+# See the blogging docs at https://github.com/pulumi/docs/blob/master/BLOGGING.md.
+# for additional details, and please remove these comments before submitting for review.
+---
+
+Stop writing Kubernetes YAML by hand, and start using the power of real programming languages! Pulumi now allows you to
+generate Kubernetes manifests that can be easily integrated into existing CI/CD workflows.
+
+<!--more-->
+
+While Pulumi has great support for deploying and updating Kubernetes resources on a cluster, many users have asked for
+the option to render YAML that they can integrate into existing workflows. The `v1.5.4` release of `pulumi-kubernetes`
+adds the `renderYamlToDirectory` option, which enables this behavior.
+
+This option is available in every Pulumi-supported language, including TypeScript/JavaScript, Python, and .NET
+([Go support is coming soon!](https://github.com/pulumi/pulumi-kubernetes/issues/70)). Aside from easily templating
+configuration across resources, using a real programming language allows you to write and consume libraries, and easily
+mix in infrastructure configuration (e.g., managed database endpoints, object storage, etc.), all in the same program.
+
+## Rendering YAML from TypeScript
+
+First, choose a directory for the rendered manifests, and specify that path on a `Provider`.
+
+```typescript
+// Instantiate a Kubernetes Provider and specify the render directory.
+const provider = new k8s.Provider("render-yaml", {
+    renderYamlToDirectory: "rendered"
+});
+```
+
+Next, use that `Provider` for any Kubernetes resources you want to render as YAML.
+
+```typescript
+// Create a Kubernetes PersistentVolumeClaim.
+const pvc = new kx.PersistentVolumeClaim("data", {
+    spec: {
+        accessModes: [ "ReadWriteOnce" ],
+        resources: { requests: { storage: "1Gi" } }
+    }
+}, { provider });
+
+// Create a Kubernetes ConfigMap.
+const cm = new kx.ConfigMap("cm", {
+    data: { "config": "very important data" }
+}, { provider });
+
+// Create a Kubernetes Secret.
+const secret = new kx.Secret("secret", {
+    stringData: {
+        "password": new random.RandomPassword("pw", {
+            length: 12}).result,
+    }
+}, { provider });
+
+// Define a Pod.
+const pb = new kx.PodBuilder({
+    containers: [{
+        env: {
+            CONFIG: cm.asEnvValue("config"),
+            PASSWORD: secret.asEnvValue("password"),
+        },
+        image: "nginx",
+        ports: {http: 8080},
+        volumeMounts: [ pvc.mount("/data") ],
+    }]
+});
+
+// Create a Kubernetes Deployment.
+const deployment = new kx.Deployment("nginx", {
+    spec: pb.asDeploymentSpec( { replicas: 3 })
+}, { provider });
+
+// Create a Kubernetes Service.
+const service = deployment.createService({type: kx.types.ServiceType.LoadBalancer});
+```
+
+Now, run `pulumi update`, and Pulumi renders these resources to YAML. The update process resolves outputs as usual,
```

Comment:
s/outputs/Outputs and lets link to [output docs](https://www.pulumi.com/docs/intro/concepts/programming-model/#outputs)